services/navigation/builtin/builtin.go

```diff
@@ -581,5 +624,81 @@ func (svc *builtIn) Obstacles(ctx context.Context, extra map[string]interface{})
 func (svc *builtIn) Paths(ctx context.Context, extra map[string]interface{}) ([]*navigation.Path, error) {
 	svc.mu.RLock()
 	defer svc.mu.RUnlock()
-	return nil, errUnimplemented
+
+	rawExecutionWaypoint := svc.activeExecutionWaypoint.Load()
+	// If there is no execution, return empty paths
+	if rawExecutionWaypoint == nil || rawExecutionWaypoint == emptyExecutionWaypoint {
+		return []*navigation.Path{}, nil
+	}
+
+	ewp, ok := rawExecutionWaypoint.(executionWaypoint)
+	if !ok {
+		return nil, errors.New("execution corrupt")
+	}
+
+	ph, err := svc.motionService.PlanHistory(ctx, motion.PlanHistoryReq{
+		ComponentName: svc.base.Name(),
+		ExecutionID:   ewp.executionID,
+		LastPlanOnly:  true,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	geoPoints := make([]*geo.Point, 0, len(ph[0].Plan.Steps))
+	for _, s := range ph[0].Plan.Steps {
+		if len(s) > 1 {
+			return nil, errors.New("multi component paths are unsupported")
+		}
+		var pose spatialmath.Pose
+		for _, p := range s {
+			pose = p
+		}
+
+		geoPoint := geo.NewPoint(pose.Point().Y, pose.Point().X)
+		geoPoints = append(geoPoints, geoPoint)
+	}
+
+	path, err := navigation.NewPath(ewp.waypoint.ID.String(), geoPoints)
```

Comment:
Seems like it would be better to keep this as they type of waypoint.ID rather than storing it as a string

Reply:
done